apps/prairielearn/src/lib/server-jobs-2.ts

```diff
@@ -0,0 +1,184 @@
+import AnsiUp from 'ansi_up';
+import execa = require('execa');
+import { z } from 'zod';
+import * as Sentry from '@prairielearn/sentry';
+import { logger } from '@prairielearn/logger';
+import { loadSqlEquiv, queryAsync, queryValidatedOneRow } from '@prairielearn/postgres';
+
+import { chalk } from './chalk';
+import serverJobs = require('./server-jobs');
+import socketServer = require('./socket-server');
+
+const sql = loadSqlEquiv(__filename);
+
+interface CreateServerJobOptions {
+  courseId?: string;
+  courseInstanceId?: string;
+  courseRequestId?: string;
+  assessmentId?: string;
+  userId?: string;
+  authnUserId?: string;
+  type: string;
+  description: string;
+}
+
+interface ServerJobExecOptions {
+  cwd?: string;
+  env?: NodeJS.ProcessEnv;
+}
+
+type ServerJobContext = Omit<ServerJob, 'execute' | 'executeInBackground'>;
+type JobExecutionFunction = (context: ServerJobContext) => Promise<void>;
+
+class ServerJob {
+  public jobSequenceId: string;
+  public jobId: string;
+  private started = false;
+  private finished = false;
+  public output = '';
+
+  constructor(jobSequenceId: string, jobId: string) {
+    this.jobSequenceId = jobSequenceId;
+    this.jobId = jobId;
+  }
+
+  error(msg: string) {
+    this.addToOutput(msg + '\n');
+  }
+
+  warn(msg: string) {
+    this.addToOutput(msg + '\n');
+  }
+
+  info(msg: string) {
+    this.addToOutput(msg + '\n');
+  }
+
+  verbose(msg: string) {
+    this.addToOutput(msg + '\n');
+  }
+
+  async exec(file: string, args: string[] = [], options: ServerJobExecOptions = {}): Promise<void> {
+    const msg = `$ ${file} ${args.join(' ')}\n`;
+    this.addToOutput(chalk.greenBright(msg));
```

Comment:
`greenBright` looked nice, so I went with it without much thought:

<img width="1086" alt="Screenshot 2023-05-02 at 17 32 38" src="https://user-images.githubusercontent.com/1476544/235813021-b516055a-b1c3-4139-b527-0ff14604f637.png">

Open to other suggestions though!

Reply:
Looks good to me!